### DISCUSSION TITLE
Unable to use AWS Credential in github action

### DISCUSSION THREAD

**Discussion by wcorbett236** [gruntwork-io/knowledge-base#706] [Help]
Hi there,  I am trying to run terragrunt init from a github action pipeline and it is just not wanting to find the AWS Key/Secret env vars I have added.  I have tried all ways of explicitly passing them but still getting no credential found error.  

<img width="1341" alt="image" src="https://user-images.githubusercontent.com/111913063/236312917-47b2ff8f-1ae9-45fc-a7eb-0f3fe45613e5.png">
I am even setting the values directly on the action step as env vars and still says not there.  

As a checkpoint I was able to pass this secret to the aws configure credentials action and do basic cli commands like list buckets 

<img width="993" alt="image" src="https://user-images.githubusercontent.com/111913063/236313212-d9edd84e-ca2d-4285-99a1-c1c236b58321.png">

---

<ins datetime="2023-05-04T19:46:10Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110145">Tracked in ticket #110145</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
I think can be multiple root causes:
* not specified AWS_REGION / AWS_DEFAULT_REGION values
* multiple AWS profiles are in use
* AWS account requires SSO configuration

to get more insights, will be helpful to have output with `--terragrunt-log-level debug` to see at which part it is failing
